strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/Maven2LayoutProvider.java

```diff
@@ -185,6 +186,8 @@ public void delete(String storageId,
             // Otherwise, this is either not an artifact directory, or not a valid Maven artifact
         }
 
+        closeIndex(repositoryPath);
```

Comment:
If you move it outside the provider, where will you put it? The closing of the index belongs to the Maven implementation.